src/csharp/Grpc.IntegrationTesting/ExternalDnsWithTracingClientServerTest.cs

```diff
@@ -0,0 +1,171 @@
+#region Copyright notice and license
+
+// Copyright 2015 gRPC authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#endregion
+
+using System;
+using System.Collections.Generic;
+using System.Net.Sockets;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Grpc.Core;
+using Grpc.Core.Logging;
+using Grpc.Core.Utils;
+using Grpc.Testing;
+using NUnit.Framework;
+
+namespace Grpc.IntegrationTesting
+{
+    /// <summary>
+    /// See https://github.com/grpc/issues/18074, this test is meant to
+    /// try to trigger the described bug.
+    /// Runs interop tests in-process, with that client using a target
+    /// name that using a target name that triggers interaction with
+    /// external DNS servers (even though it resolves to the in-proc server).
+    /// </summary>
+    public class ExternalDnsWithTracingClientServerTest
+    {
+        Server server;
+        Channel channel;
+        TestService.TestServiceClient client;
+        SocketUsingLogger newLogger;
+
+        [OneTimeSetUp]
+        public void Init()
+        {
+            Environment.SetEnvironmentVariable("GRPC_TRACE", "all");
+            Environment.SetEnvironmentVariable("GRPC_VERBOSITY", "DEBUG");
+            newLogger = new SocketUsingLogger(GrpcEnvironment.Logger);
+            Assert.AreEqual(newLogger.LogCount, 0);
```

Comment:
Because of the note above, I'm afraid that the logCount might just be 0 because the of setting GRPC_TRACE and GRPC_VERBOSITY actually isn't effective on Unix systems.
I know that we are primarily trying to test things on windows, but I think the assertion is misleading anyway.

Reply:
I've removed this sanity checking. It seems to be working on linux, just since the tests are passing on linux, however I'm good to leave out the check since there are known open issues.

Since this test is primarily intended for Windows (though it could potentially also be interesting to run on linux), I've just added a `TODO` to delegate the fixing of this test on linux to the solution for #14963. Alternatively, we could perhaps set these env vars in `run_tests.py`, but I'm not sure it's worth doing that yet.